crates/cargo-bsp/src/server/dispatch.rs

```diff
@@ -77,13 +77,21 @@ impl<'a> RequestDispatcher<'a> {
             None => return self,
         };
         let sender_to_main = self.global_state.handlers_sender.clone();
-        let request_handle = RequestHandle::spawn::<R>(
+        match RequestHandle::spawn::<R>(
             Box::new(move |msg| sender_to_main.send(msg).unwrap()),
             req.id.clone(),
             params,
             self.global_state.config.root_path(),
-        );
-        self.global_state.handlers.insert(req.id, request_handle);
+        ) {
```

Comment:
Zostaw może zmienną request_handle i na niej zrób match, bo aktualnie dość ciężko się to czyta